### PR TITLE
integration test table search

### PR DESCRIPTION
**Pull request by callunity**
GLT-1954

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernatePoolableElementViewDao.java#L28](https://github.com/TGAC/miso-lims/blob/7f98e26262422357f04695d208b596f37e3ef877/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolableElementViewDao.java#L28): Define a constant instead of duplicating this literal "dilutionBarcode" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernatePoolableElementViewDao.java#L28](https://github.com/TGAC/miso-lims/blob/88748b5b93739c69aef53bcb5acbf1e42166ae33/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolableElementViewDao.java#L28): Define a constant instead of duplicating this literal "dilutionBarcode" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/ListTablesIT.java [hunk 1]
```diff
@@ -358,22 +522,47 @@ private void testTabbedColumnsSort(String listTarget) {
...
-      assertTrue("clicked tab without errors", LimsUtils.isStringEmptyOrNull(page.getErrors().getText()));
+      assertTrue("clicked tab without errors", isStringEmptyOrNull(page.getErrors().getText()));
     });
   }
 
+  private void testSearch(String listTarget, String searchTarget) {
+    // confirm that searching by a term returns no errors
+    ListPage page = getList(listTarget);
+    page.getTable().searchFor(searchTarget);
+    assertTrue("error searching " + listTarget + " page for '" + searchTarget + "': ", isStringEmptyOrNull(page.getErrors().getText()));
```
Comment (djcooke) [inline miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/ListTablesIT.java]: This is testing that the page doesn't politely inform you of a failure, but not that the search actually worked as expected. Should be testing that an expected record was found, and an unexpected record wasn't. I don't think there's a one-size-fits-all method for this

Comment (callunity) [reply miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/ListTablesIT.java]: As per our conversation, I've made a ticket to write unit tests for the Pagination Filters, as unit tests will run faster and can be less brittle.

Comment (djcooke) [reply miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/ListTablesIT.java]: :+1:

**Pull request merged**